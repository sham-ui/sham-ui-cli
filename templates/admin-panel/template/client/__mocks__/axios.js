const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
