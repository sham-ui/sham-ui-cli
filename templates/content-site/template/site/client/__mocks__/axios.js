const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
