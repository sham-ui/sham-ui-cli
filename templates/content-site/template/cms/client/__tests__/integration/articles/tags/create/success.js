import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
} );

it( 'success create tag', async() => {
    expect.assertions( 5 );

    const DI = setup();
    axios
        .useDefaultMocks()
        .use( 'get', '/tags', {
            tags: []
        } );

    history.pushState( {}, '', 'http://client.example.com/articles/tags' );
    await app.start( DI );

    app.click( '[data-test-toggle-create-form]' );
    await app.waitRendering();

    app.checkMainPanel();

    const formData = {
        name: 'Travel'
    };

    axios
        .use( 'post', '/tags', {} )
        .use( 'get', '/tags', {
            tags: [
                {
                    id: 1,
                    name: formData.name,
                    slug: 'travel'
                }
            ]
        } );

    app.form.fill( 'name', formData.name );
    app.form.submit();

    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();
    app.checkMainPanel();

    expect( axios.mocks.post ).toHaveBeenCalledTimes( 1 );
    expect( axios.mocks.post.mock.calls[ 0 ][ 0 ] ).toBe( '/tags' );
    expect( axios.mocks.post.mock.calls[ 0 ][ 1 ] ).toEqual( formData );
} );
