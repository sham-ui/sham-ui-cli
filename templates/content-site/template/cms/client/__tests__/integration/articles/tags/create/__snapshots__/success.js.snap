// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success create tag 1`] = `
"<h4 class=\\"page-header\\"> Tags
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"crud-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Hide create new tag form</button>
      <div class=\\"panel mg-b-30 item-create\\">
        <form class=\\"form-layout\\">
          <!--0-->
          <!--0-->
          <!--1-->
          <div class=\\"form-group\\">
            <label>Name:</label>
            <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
          </div>
          <!--0-->
          <!--2-->
          <button type=\\"submit\\">
            <!--3-->Create
            <!--4-->
          </button>
          <!--5-->
        </form>
        <!--0-->
        <!--1-->
        <!--0-->
        <!--0-->
        <!--0-->
        <!--0-->
      </div>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"items-list\\">
        <div class=\\"panel\\">
          <table>
            <thead>
              <tr>
                <th>ID</th>
                <th>Name</th>
                <th>Slug</th>
                <th></th>
              </tr>
            </thead>
            <tbody></tbody>
          </table>
          <!--0-->
          <!--0-->
          <!--0-->
          <!--1-->
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->"
`;

exports[`success create tag 2`] = `
"<h4 class=\\"page-header\\"> Tags
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"crud-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Show create new tag form</button>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"items-list\\">
        <div class=\\"panel\\">
          <table>
            <thead>
              <tr>
                <th>ID</th>
                <th>Name</th>
                <th>Slug</th>
                <th></th>
              </tr>
            </thead>
            <tbody>
              <tr class=\\"\\">
                <td class=\\"text-center\\">1</td>
                <td>Travel</td>
                <td>travel</td>
                <td class=\\"actions\\">
                  <button data-test-update-button=\\"1\\"> Edit </button>
                  <button class=\\"btn-danger\\" data-test-delete-button=\\"1\\"> Delete </button>
                </td>
              </tr>
            </tbody>
          </table>
          <!--0-->
          <!--0-->
          <!--0-->
          <!--1-->
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->
<!--0-->"
`;
